extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java

```diff
@@ -151,15 +147,17 @@
 
     private static final Logger LOG = Logger.getLogger(HibernateOrmProcessor.class);
 
-    private static final DotName TENANT_CONNECTION_RESOLVER = DotName.createSimple(TenantConnectionResolver.class.getName());
-    private static final DotName TENANT_RESOLVER = DotName.createSimple(TenantResolver.class.getName());
-
-    private static final DotName STATIC_METAMODEL = DotName.createSimple(StaticMetamodel.class.getName());
-    private static final DotName PERSISTENCE_UNIT = DotName.createSimple(PersistenceUnit.class.getName());
-    private static final DotName PERSISTENCE_UNIT_REPEATABLE_CONTAINER = DotName
-            .createSimple(PersistenceUnit.List.class.getName());
-    private static final DotName JPA_ENTITY = DotName.createSimple(Entity.class.getName());
-    private static final DotName MAPPED_SUPERCLASS = DotName.createSimple(MappedSuperclass.class.getName());
+    public static final DotName TENANT_CONNECTION_RESOLVER = DotName
+            .createSimple("io.quarkus.hibernate.orm.runtime.tenant.TenantConnectionResolver");
+    public static final DotName TENANT_RESOLVER = DotName
+            .createSimple("io.quarkus.hibernate.orm.runtime.tenant.TenantResolver");
+
+    public static final DotName STATIC_METAMODEL = DotName.createSimple("javax.persistence.metamodel.StaticMetamodel");
+    public static final DotName PERSISTENCE_UNIT = DotName.createSimple("io.quarkus.hibernate.orm.PersistenceUnit");
+    public static final DotName PERSISTENCE_UNIT_REPEATABLE_CONTAINER = DotName
+            .createSimple("io.quarkus.hibernate.orm.PersistenceUnit$List");
+    public static final DotName JPA_ENTITY = DotName.createSimple("javax.persistence.Entity");
+    public static final DotName MAPPED_SUPERCLASS = DotName.createSimple("javax.persistence.MappedSuperclass");
```

Comment:
The main problem I have with this strategy is that anyone changing this code later will likely add a constant and forget about testing it. So these tests you're adding are shielding us from problems for these constants, but not for future ones.

One idea to make this future-proof is to regroup constants in a single class that has a specific way of building the constants, e.g.:

```java

public class ClassNames {
    public static final DotName TENANT_CONNECTION_RESOLVER = createConstant("io.quarkus.hibernate.orm.runtime.tenant.TenantConnectionResolver");
    public static final DotName TENANT_RESOLVER = createConstant("io.quarkus.hibernate.orm.runtime.tenant.TenantResolver");

    public static final DotName STATIC_METAMODEL = createConstant("javax.persistence.metamodel.StaticMetamodel");
    public static final DotName PERSISTENCE_UNIT = createConstant("io.quarkus.hibernate.orm.PersistenceUnit");
    public static final DotName PERSISTENCE_UNIT_REPEATABLE_CONTAINER = createConstant("io.quarkus.hibernate.orm.PersistenceUnit$List");
    public static final DotName JPA_ENTITY = createConstant("javax.persistence.Entity");
    public static final DotName MAPPED_SUPERCLASS = createConstant("javax.persistence.MappedSuperclass");

    private ClassNames() {
    }

    static final Set<DotName> CREATED_CONSTANTS = new HashSet<>();

    private static DotName createConstant(String fqcn) {
        DotName result = DotName.createSimple(fqcn);
        CREATED_CONSTANTS.add(result);
        return result;
    }

}
```

Then when someone adds a constant, hopefully by copy-pasting one of the existing constants, it will automatically get tested too, thanks to this more generic test:

```java
public class ConstantsTest {

    @ParameterizedTest
    @MethodSource("provideConstantsToTest")
    void testClassNameRefersToExistingClass(DotName constant) {
          Assertions.assertThatCode(() -> getClass().getClassLoader().loadClass(constant.toString()))
              .doesNotThrowAnyException();
    }

    private Set<DotName> provideConstantsToTest() {
        return ClassNames.CREATED_CONSTANTS;
    }

}
````

Reply:
It's a nice idea but I see no way to ensure everyone follows such a pattern.

Also TBH I started this work with a "Constants" class, but then I backed off as it was getting complicated, and I don't see a strong need to have all the strings tested. The tests I introduced are going to be helpful as a reminder if e.g. ORM6 were to change its internal package names, but a single reminder should be effective enough to point at the general issue.

Reply:
> It's a nice idea but I see no way to ensure everyone follows such a pattern.

I agree we cannot be completely fool proof without some very heavy tooling, but I think we could try a bit more.

My point was that at least this way, someone adding a new constant *may* follow the same pattern, because they'll start from an existing constant, see that constants are all in a specific place and created in a specific way.

In the current state, I'm fairly certain these tests will be forgotten about and new constants will be introduced without testing them: we'll follow the same pattern as everywhere (hard-coded strings) because we have to, but not everyone will remember to test the strings.

> I don't see a strong need to have all the strings tested

I think the point of your change was "it's no big deal to use hard-coded constants as long as we have tests"? Well, if we don't test the constants, it becomes a big deal in my opinion.

I wish I was certain that the correctness of every constant was checked indirectly by unit tests or integration tests, but... well, I'm not :/

Reply:
Fine sure let's try a bit more. One warning though: I will have ZERO time for this this week; up to you to send an improvement proposal.

One idea could be to use Forbidden API itself to "ban" creation of `DotName` outside of such a selected blessed utility, and same for the BuildItem constructor which takes a string. Not sure if there's many other use cases or how inconvenient maintenance would become though; I would suggest try but consider backing off if you see it getting out of reason.